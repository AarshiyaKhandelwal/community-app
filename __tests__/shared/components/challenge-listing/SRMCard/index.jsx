import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import SRMCard from 'components/challenge-listing/SRMCard';

const mockDatas = [
  {
    category: 'now',
    srmChallenge: {},
  }, {
    category: 'upcoming',
    srmChallenge: {},
  }, {
    category: 'past',
    srmChallenge: {},
  },
];

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <SRMCard {...this.props} />;
  }
}

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <SRMCard {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    TU.renderIntoDocument((<Wrapper {...data} />));
  });
});

