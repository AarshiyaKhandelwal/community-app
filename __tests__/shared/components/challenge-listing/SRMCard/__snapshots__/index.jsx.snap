// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="SRMCard-container"
>
  <HappeningNow />
  
  
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="SRMCard-container"
>
  
  <UpcomingSRMs
    srmChallenge={Object {}}
  />
  
</div>
`;

exports[`Matches shallow shapshot 3`] = `
<div
  className="SRMCard-container"
>
  
  
  <PastSRMs />
</div>
`;
